vagrant/plugins/vagrant-hypconfigmgmt/spec/unit/command/ensure_default_domain_configured_spec.rb

```diff
@@ -0,0 +1,57 @@
+# -*- encoding: utf-8 -*-
+# vim: set fileencoding=utf-8
+
+require 'spec_helper'
+require "vagrant-hypconfigmgmt/command"
+
+describe VagrantHypconfigmgmt::Command do
+  # create a fake app and env to pass into the VagrantHypconfigmgmt::Command constructor
+  let(:app) { }
+  let(:env) { { :ui => ui } }
+
+  # pretend env contains the Vagrant ui element
+  let(:ui) do
+    double('ui').tap do |ui|
+      allow(ui).to receive(:info) { nil }
+    end
+  end
+
+  # Call the method under test after every 'it'. Similar to setUp in Python TestCase
+  after do
+    subject.ensure_default_domain_configured(env)
+  end
+
+  # instantiate class of which a method is to be tested
+  subject { described_class.new(app, env) }
+
+  # the method that we are going to test
+  describe "#ensure_default_domain_configured" do
+
+    context "when a default domain is configured" do
+      let(:retrieved_settings) { { "hostmanager" => { "default_domain" => "exaxmple.com" } } }
+      it "does not change the retrieved settings" do
```

Comment:
this test tests that if a value is configured it is not overwritten with the default and the retrieved settings remain unchanged. we only need to mutate it in case of missing values, then we set the default default_domain (hypernode.local)